src/pages/ProductListing/components/ProductCard/ProductCard.jsx

```diff
@@ -1,48 +1,107 @@
 import { useCart } from "../../../../context/cartContext";
-import { actionTypes } from "../../../../reducers/actionTypes";
-import { addToCartService } from "../../../../services/addToCartService";
+import { addToCartService } from "../../../../services/cart-services";
+import {
+	addToWishlistService,
+	removeProductWishlistService,
+} from "../../../../services/wishlist-services";
 import { useAuth } from "../../../../context/AuthContext";
 import { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
+import { useWishlist } from "../../../../context/WishlistContext";
+
 export const ProductCard = ({ product }) => {
-	const { cartState, cartDispatch } = useCart();
-	const { SET_CART } = actionTypes;
+	const { cart, setCart } = useCart();
 	const { auth } = useAuth();
 	const [inCart, setInCart] = useState(false);
+	const [inWishlist, setInWishlist] = useState(false);
 	const addToCartServerCall = async () => {
-		const cart = await addToCartService(product, auth.token);
-		if (cart) cartDispatch({ type: SET_CART, payload: { cart } });
+		try {
```

Comment:
Not a necessity, but is it possible to move these calls to a separate file? If yes, then you can try doing it!